### PR TITLE
Add myself to CODEOWNERS

### PR DESCRIPTION
**Pull request by nikic**
Automatically add myself as a reviewer for some of the passes I maintain.

### CODE REVIEW

**Comment by nikic** [comment]
@tstellar @MaskRay Just want to double check that it's okay to use this file now that the subscriber teams have been removed from it.

**Comment by tstellar** [comment]
I think we need some kind of general policy on how to use this file.  It  might be good to start with a Discourse post about it.

**Comment by MaskRay** [comment]
> I think we need some kind of general policy on how to use this file. It might be good to start with a Discourse post about it.

I am totally happy if you want to own these files (and you've already made many good changes to them) but I agree that a post is needed about the usage.

Currently there is [`pr-subscribers-llvm-analysis`](https://github.com/orgs/llvm/teams/pr-subscribers-llvm-analysis) but no fine-grained team for a subset of files. It seems that if you want to act more like a gatekeeper of these files you will have to follow the email notifications more closely...

**Comment by nikic** [comment]
Discourse thread: https://discourse.llvm.org/t/usage-of-codeowners-file/73524

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,4 +1,27 @@
...
+# reviewing a large fraction of pull requests in the relevant area.
+#
+# The GitHub "code owners" mechanism is used exclusively to auto-assign
+# reviewers and does not carry significance beyond that. It is not necessary
+# to receive an approval from a "code owner" in particular -- any LLVM project
+# member can approve pull requests.
+#
+# Note that GitHub's concept of "code owner" is independent from LLVM's own
+# "code owner" concept, they merely happen to share terminology. See the
+# CODE_OWNERS.txt files in the respective subproject directories.
```
Comment (asb) [inline .github/CODEOWNERS:12]: I'd additionally link to https://llvm.org/docs/DeveloperPolicy.html#code-owners

**Review by joker-eph — APPROVED** [review]
LGTM but this probably deserve to get more approval, @tstellar in particular?

**Review by tstellar — APPROVED** [review]

**Review by asb — APPROVED** [review]

**Pull request merged**